### PR TITLE
Rho Diags: Do Not Allocate `rho_fp/cp`, Remove `plot_rho`, `setplot_rho`

### PR DESCRIPTION
**Pull request by RemiLehe**
This saves a bit of memory. Also removes `plot_rho` and `setplot_rho` which are not used anymore in the code.

### CODE REVIEW

**Review by EZoni — APPROVED** [review]
Thanks, Remi! Looks good to me, will merge as soon as CI is green.

**Comment by EZoni** [comment]
As discussed on Slack, the issue with the test `LaserAccelerationBoost`, the only one currently failing, is probably this code block here, which should be removed:
https://github.com/ECP-WarpX/WarpX/blob/eb7d5d966a4a4ac8eb84d673d50a213c8e618eec/Source/Diagnostics/FullDiagnostics.cpp#L425-L435

File: Source/Diagnostics/FullDiagnostics.cpp [hunk 1]
```diff
@@ -422,21 +422,8 @@ FullDiagnostics::InitializeFieldFunctors (int lev)
...
-                    m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_rho_fp(lev),
-                                                                                          lev, m_crse_ratio);
-                }
-            }
-            else {
-                // Initialize rho functor to dump total rho
-                m_all_field_functors[lev][comp] = std::make_unique<RhoFunctor>(lev, m_crse_ratio);
-            }
+            // Initialize rho functor to dump total rho
+            m_all_field_functors[lev][comp] = std::make_unique<RhoFunctor>(lev, m_crse_ratio);
```
Comment (EZoni) [inline Source/Diagnostics/FullDiagnostics.cpp:426]: @RevathiJambunathan Do you agree that this is now correct?

Comment (RevathiJambunathan) [reply Source/Diagnostics/FullDiagnostics.cpp:426]: Yes @EZoni Looks good to me!

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -1491,10 +1491,9 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     }
 #endif
 
-    bool deposit_charge = do_dive_cleaning || (plot_rho && do_back_transformed_diagnostics);
+    bool deposit_charge = do_dive_cleaning;
```
Comment (RevathiJambunathan) [inline Source/WarpX.cpp:1494]: If plot_rho is not used anymore, we can also remove its definition from WarpX.H
and wherever setplot_rho() is defined and/or called

**Review by RevathiJambunathan — COMMENTED** [review]
Thank you for the PR! Looks good to me. A very minor clean-up suggestion.

File: Source/WarpX.cpp (hunk 2 above)
Comment (EZoni) [reply Source/WarpX.cpp:1494]: Good point, thanks Reva! I will check if we can remove `plot_rho` and associated functions from the code. Update: it seems like we can remove both, see https://github.com/ECP-WarpX/WarpX/pull/2203/commits/07ea5535e167529d5271de459e4b75f4e939765c.

**Review by RevathiJambunathan — APPROVED** [review]

**Pull request merged**